### ISSUE TITLE
Margin around calendar view leads to dithering when scrolling

### ISSUE THREAD

**Issue by sedrubal** [Etar-Group/Etar-Calendar#457]
When you zoom the calendar view very small and your screen is large enough, there is a small margin around the calendar view at the top and bottom. When you scroll, there are ugly dithering effects, like you can see in this screen record:
https://i.imgur.com/sZewz8T.mp4

Phone: OnePlus 5 with Andoid 9 (default display size, 5.5 inches, 1080 x 1920 pixels, 16:9 ratio (~401 ppi density).

**Comment by Gitsaibot** [comment]
Maybe #rparkins999 can have a look at that. He worked on it a few years ago.

**Comment by jspricke** [comment]
This was done in #197, @rparkins999 can you have a look maybe?

(Just restating this to make sure the notifications actually works.)

**Comment by rparkins999** [comment]
I don't see any dithering on my system (Galaxy S6 running Android 7.0). Jspricke, are you caliming that that problem doesn't occur before my pull request, and does occur after it? I don't think I've updated my copy of Etar since that change (it reports build version 1.0.10), so it may be a more recent issue.

**Comment by nottheend** [comment]
I can also confirm the dithering:

Zooming the calendar very small in the week-view, it starts dithering in an ugly way.

Phone: Galaxy A7, Android 9

**Comment by SebiderSushi** [comment]
On my device i can reproduce it as well by:
- Choosing a limited, scrollable and zoomable calendar view, i.e weekly or dayly
- Zooming out as far as possible (all 24 hours are on screen)
- While touching the screen to scroll, the content rapidly jumps back and forth between two positions.

I can see the little excess margins added at the top and the bottom. That is, when zoomed in a little, scrolling is limited at exactly 00:00 or 24:00 respectively - as represented by the lines separating each hour. Once the view has been zoomed out completely to show all 24 hours of the day, "half an hour" of UI space is added each above 00:00 and below 24:00, so that the scrollable UI takes up 25:00 in total.

The smallest zoom level allows exactly 24:30 to be displayed on my screen. Once a finger touches the screen any microscopic scrolling movement up or down lets the UI tip between "top" and "bottom" position, thus jumping back and forth along that additional half hour that won't fit the screen, as if some logic would try to align both the upper and the lower end with the viewframe boundaries even though the contents are a little bit larger.

As it appears on my device, there must be two solutions:
- Disable scrolling when the daily or weekly views are considered to be zoomed out completely
- Ensure that the UI never adds those additional half hours to the 24 hour view on any possible screen or device

In either case, this might be a good place to start since this additional UI space must somehow be added programatically at some point...
And maybe my description of the whole thing managed to get some bells ringing for anyone familiar with the implementation?

Otherwise, if this is still not reproducable by any developer i can try to make and post some other screen recordings as well or possibly do some deeper debugging & code testing.

Etar version is 1.0.21 from F-Droid.

**Comment by rparkins999** [comment]
As previously stated, my phone reports v1.0.10, but this is because I built the etar on it from my own tree (and didn't increase the version number) rather than downloading from fdroid after my change was integrated. The first published release including my change was v1.0.11. v1.0.12, which according to SebiderSushi first shows the problem, seems to only include some changes to descriptive text and no changes to the code.
On my phone if I compress the week view so that the whole 24 hours is visible, and try to scroll up or down, I see a bit of jumping on the screen as long as I'm actually pushing with my finger, but it stabilises as soon as I take my finger off the screen. If that is the problem reported, it doesn't seem to be very serious, but it does seem to be the result of my change. I'll upgrade my fork to the current released version (1.0.21) and build it and take a look at it as soon as I have time.

**Comment by rparkins999** [comment]
I've just cloned the latest version, and I get lots of errors trying to build it with the current version of Android Studio. I can't really investigate the problem without a build and debug environment, and I'm not willing to make a whole lot of changes unrelated to this issue in order to get it to build.

**Comment by Gitsaibot** [comment]
The master should run fine with the latest android studio version. I use it too and have no build errors.

**Comment by rparkins999** [comment]
~/Documents/Android> git clone https://github.com/Etar-Group/Etar-Calendar.git
Cloning into 'Etar-Calendar'...
remote: Enumerating objects: 4, done.
remote: Counting objects: 100% (4/4), done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 42967 (delta 0), reused 0 (delta 0), pack-reused 42963
Receiving objects: 100% (42967/42967), 35.05 MiB | 891.00 KiB/s, done.
Resolving deltas: 100% (22535/22535), done.
~/Documents/Android> ~/Android/Sdk/cmdline-tools/latest/bin/sdkmanager platform-tools export ANDROID_HOME=~/Android/Sdk git submodule update --init gradle build
Warning: Unknown argument --init
Warning: Could not create settings
java.lang.IllegalArgumentException
        at com.android.sdklib.tool.sdkmanager.SdkManagerCliSettings.<init>(SdkManagerCliSettings.java:419)
        at com.android.sdklib.tool.sdkmanager.SdkManagerCliSettings.createSettings(SdkManagerCliSettings.java:152)
        at com.android.sdklib.tool.sdkmanager.SdkManagerCliSettings.createSettings(SdkManagerCliSettings.java:134)
        at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:57)
        at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:48)
Usage:
  sdkmanager [--uninstall] [<common args>] [--package_file=<file>] [<packages>...]
  sdkmanager --update [<common args>]
  sdkmanager --list [<common args>]
  sdkmanager --licenses [<common args>]
  sdkmanager --version

With --install (optional), installs or updates packages.
    By default, the listed packages are installed or (if already installed)
    updated to the latest version.
With --uninstall, uninstall the listed packages.

    <package> is a sdk-style path (e.g. "build-tools;23.0.0" or
             "platforms;android-23").
    <package-file> is a text file where each line is a sdk-style path
                   of a package to install or uninstall.
    Multiple --package_file arguments may be specified in combination
    with explicit paths.

With --update, all installed packages are updated to the latest version.

With --list, all installed and available packages are printed out.

With --licenses, show and offer the option to accept licenses for all
     available packages that have not already been accepted.

With --version, prints the current version of sdkmanager.

Common Arguments:
    --sdk_root=<sdkRootPath>: Use the specified SDK root instead of the SDK
                              containing this tool

    --channel=<channelId>: Include packages in channels up to <channelId>.
                           Common channels are:
                           0 (Stable), 1 (Beta), 2 (Dev), and 3 (Canary).

    --include_obsolete: With --list, show obsolete packages in the
                        package listing. With --update, update obsolete
                        packages as well as non-obsolete.

    --no_https: Force all connections to use http rather than https.

    --proxy=<http | socks>: Connect via a proxy of the given type.

    --proxy_host=<IP or DNS address>: IP or DNS address of the proxy to use.

    --proxy_port=<port #>: Proxy port to connect to.

    --verbose: Enable verbose output.

* If the env var REPO_OS_OVERRIDE is set to "windows",
  "macosx", or "linux", packages will be downloaded for that OS.
~/Documents/Android>

**Comment by Gitsaibot** [comment]
> ~/Documents/Android> /Android/Sdk/cmdline-tools/latest/bin/sdkmanager platform-tools export ANDROID_HOME=/Android/Sdk git submodule update --init gradle build

Do you use android studio on windows ? You have to run git submodule update --init within the project directory(Etar-Calendar).

**Comment by rparkins999** [comment]
No, I'm running Linux. Running the command in the project directory gets the same error. Are you using the latest version of command line SDK tools as installed by Android Studio?

**Comment by jspricke** [comment]
> git submodule update --init gradle build

those are two commands

> Warning: Unknown argument --init

that sounds wrong, git should work

**Comment by Gitsaibot** [comment]
> Are you using the latest version of command line SDK tools as installed by Android Studio?

I use android studio on Linux too. SDK tools = 26.1.1; SDK Platform Tools = 30.0.1 and Android Studio 3.6.3

**Comment by rparkins999** [comment]
I was eventually able to build a cloned copy of  Etar-Group/Etar-Calendar, but that doesn't quite do it because I need to build in my own fork so that I can issue a pull request if I can find a fix for the issue. Merging the current upstream version into my fork produced masses of errors in the build again, which I managed to fix by manually copying files from the clone and removing a whole lot of .orig files left around by the merge which Android Studio seemed to be trying to include in teh build. However when I try to run it on the emulator I get INSTALL_FAILED_CONFLICTING_PROVIDER. I didn't have a previous version of Etar installed on the emulator, but it has its built-in calendar app which may be causing the clash.
Removing the code
```
<provider android:name="com.android.calendar.CalendarRecentSuggestionsProvider"
            android:exported="false"
            android:authorities="com.android.calendar.CalendarRecentSuggestionsProvider" />
```
from AndroidManifest.xml fixes this problem, but I presume this code was there for a reason. It installs without this change on my phone.

**Comment by SebiderSushi** [comment]
What about forking from the master branch to a new one and fixing the issue right there?

**Comment by SebiderSushi** [comment]
Why not fix it ontop of the current master branch? I.E. fetching the upstream master branch to your local fork repo and then forking from `upstream/master` to a dedicated feature branch?

**Comment by rparkins999** [comment]
I'm not sure how SebiderSushi's suggestion would help. I would still need to fix the problems that make my (fully updated) copy of Android Studio refuse to build it, and I don't know if github will let me do a pull request from a feature branch in my fork to upstream/master. In any case if I don't merge into my own master branch it becomes an orphan which I can't do anything with any more because it will be permanently out of date with respect to upstream/master.

I'm not a github expert: I'm just trying to follow the instructions given in the git help screens to contribute to a project. If those instructions aren't appropriate for Etar-Calendar, that should have been made clear to me before I ever made the original change in 2017.

If I do a pull request from my fork's master into upstream/master, I don't see why github reports all the change history since my fork's master was originally created (as it seems to do). The only changes that should be made to upstream/master if the pull request is accepted are those that I made since I last merged from upstream/master into my fork's master 3 days ago. Those were (I think correctly) split into three commits: commit 9d745e5dfe5aba76963c50af0a65ab2085dea885 made it build; commit 36f1e6609eea42a4a8c5ea942983e577dea4a484 removed jitters and bounces when scrolling, and commit cd1ecb86bba786d5d7905319298dad1af0990375 added the additional improvement to display the hour 00 label if there is room to do so.

**Comment by jspricke** [comment]
@rparkins999 could you please stop making statements based on wrong assumptions. There is nothing special in the Etar, or even Github, workflows, it's all standard git. Feel free to ask if you have problems, we are happy to help, like I did in #457. But this is the Etar issue tracker, not a git introduction. For that I propose to read the git documentation or some tutorials.

**Comment by SebiderSushi** [comment]
While i agree with @jspricke 's statement in regards to content, i find it a bit harsh towards someone who is willing to help. On the other hand, i'm not the one managing the project so i have no idea what one has to put up with while doing so.

Now, the rest of this comment is for @rparkins999

What i'm referring to is the [feature branch workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow).
The idea of feature branches is to prevent something like your current local master branch from happening. Every feature receives its own branch that should only contain commits relevant to the feature and be based on an up to date version of the upstream branch it was forked from, since it will be merged back into this branch once the feature is complete. This ensures that after merging, the commit history of `upstream/master` remains clean, i.e. only the commits relevant for the feature are added and no merge commits created.
After that, the feature branch is supposed to be orphaned and can be deleted, since its commits are all preserved in the upstream branch.
If the upstream master branch moves on after forking to your local feature branch and starting your work, it is best to [rebase](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) the existing changes onto the current upstream master before merge.

GitHub should let you make pull requests regardless of branches.

I don't know what might have happened while you tried to merge the upstream master branch into your local master branch, but if i
```
git clone https://github.com/Etar-Group/Etar-Calendar &&
cd Etar-Calendar/ &&
git submodule update --init
```
then open the directory in android studio it builds without issues after the project has been imported and indexed completely and all build dependencies have been downloaded. So your local master branch is probably the one that got mangled a bit.
